### ISSUE TITLE
script continuous-localization workflow between GitHub repository and Weblate component

### ISSUE THREAD

**Issue by cfm** [freedomofpress/securedrop-client#1266]
After #1264, implement a workflow to run on commits to Weblate's tracking branch (see below):

0. [x] GitHub Actions workflow: nightly run
1. [x] `pybabel extract` source strings → `.pot`
    - [x] including translator comments (per #1265)
    - [x] commit
2. [x] `msgmerge` `.pot` → `.po`
    - [x] commit
3. [x] Compile `.po` → `.mo`
    - [x] When is this (automatically) committed?  →  It isn't; it's used in packaging and at development runtime.
3. [x] Store or [retrieve](https://docs.weblate.org/en/latest/api.html#get--api-components-(string-project)-(string-component)-translations-) [list of supported (100% translated) locales](https://github.com/freedomofpress/securedrop/blob/826c178f3d9e59b829c42f6afe619ee23fae45bd/securedrop/i18n_tool.py#L30-L61)
    - For SecureDrop Server, this lives in [`securedrop-docs/docs/includes/l10n.txt`](https://github.com/freedomofpress/securedrop-docs/blob/main/docs/includes/l10n.txt). 
    - This designation, and the process for granting/revoking it, might be worth formally reconsidering—though the stakes are lower on the Journalist Interface (and client) than on the Source Interface.
    - [x] commit
4. Give credit to contributing translators → #1280

Once tested manually, these operations can be scripted (either separately or as Makefile targets) initially for manual invocation, but [continuous localization](https://docs.weblate.org/en/latest/admin/continuous.html) will require running them as part of CI.  (A goal of this effort, in any case, is to avoid replicating the complexity of `securedrop`'s `i18n_tool.py`.)  In the latter case, Weblate's tracking branch will be:

1. `i18n-testing` for the purposes of this ticket; and then
2. `main` upon `i18n-testing`'s approval for merge into `main`.

If "naïve" continuous localization (committed directly to the tracking branch) turns out to be a hassle (e.g., by cluttering Git history with noisy Weblate-related commits that maintainers find annoying), we can make the workflow [asymmetric](https://docs.weblate.org/en/latest/admin/continuous.html#pushing-changes-from-weblate), in which:

1. source strings are made available for continuous translation (`main` → Weblate); but
2. translations are pushed only to ~an `i18n` branch~ a fork, from which a PR is opened for review (Weblate → fork → PR → `main`).

_Originally posted by @cfm in https://github.com/freedomofpress/securedrop-client/issues/239#issuecomment-867852782_

## [Process diagrams](https://gist.github.com/cfm/416b18a700545e63188a4daef21d599f) ([Mermaid](https://mermaid-js.github.io/mermaid-live-editor/edit#))

| Filename | Version | Description |
| --- | --- | --- |
| `weblate-r0.mmd` | 0 | Weblate canonical workflow—adapted from [their Graphviz](https://github.com/WeblateOrg/weblate/blob/main/docs/admin/continuous.rst) |
| `securedrop-client-r1.mmd` | 1 | `main` → Weblate → fork → PR → `main` |

## Subtasks

1. [x] r1 tested manually
2. [x] r1 scripted in Makefile targets

**Comment by cfm** [comment]
Updated the description to reflect my intended approach after #1264.

**Comment by cfm** [comment]
@rmol today recommended skipping even testing the "naïve" continuous-translation workflow, in which translations would be pushed from Weblate directly into `main`.  Instead, I'll start by testing Weblate's support for opening GitHub pull requests from a fork.  I'm tracking my iterative process diagrams in the gist linked at the top of this ticket.

**Comment by cfm** [comment]
While the GitHub ⟷ Weblate portions of this workflow are blocked by freedomofpress/infrastructure#3516, I've scripted the pre- and post-Weblate Babel operations in Make + GitHub Actions (at `rmol`'s recommendation) in `1266-continuous-l10n`.  The GitHub Actions portion will run nightly in a private fork for the next couple of days to kick its tires while I work/wait on the rest.

**Comment by cfm** [comment]
Noting here for the record:  freedomofpress/infrastructure#3516 appears to require relocating Weblate's SSH public key from the repository (`securedrop-i18n`) to the user (@weblate-fpf) level.  Fortunately, `i18n_tool.py` appears to be agnostic on whether translation commits, `identified by the presence of "Translated using Weblate" in the commit message`, have come from any particular GitHub user or indeed from Weblate at all:

https://github.com/freedomofpress/securedrop/blob/826c178f3d9e59b829c42f6afe619ee23fae45bd/securedrop/i18n_tool.py#L334-L343

**Comment by creviera** [comment]
https://github.com/freedomofpress/securedrop-client/pull/1282 is the PR that closes this issue and is already on the team board as ready for review so to avoid redundancy and to save a minute of time :P (always be optimizing) i'm going to move this off the board as ready to review (since it's the PR that needs reviewing)
